cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterController.java

```diff
@@ -77,19 +83,27 @@ private void setupOperations() {
         configMapOperations = new ConfigMapOperations(vertx, client);
         PvcOperations pvcOperations = new PvcOperations(vertx, client);
         deploymentOperations = new DeploymentOperations(vertx, client);
-        ImageStreamOperations imagesStreamResources;
+        ImageStreamOperations imagesStreamOperations;
         BuildConfigOperations buildConfigOperations;
         boolean isOpenShift = Boolean.TRUE.equals(client.isAdaptable(OpenShiftClient.class));
         if (isOpenShift) {
-            imagesStreamResources = new ImageStreamOperations(vertx, client.adapt(OpenShiftClient.class));
+            imagesStreamOperations = new ImageStreamOperations(vertx, client.adapt(OpenShiftClient.class));
             buildConfigOperations = new BuildConfigOperations(vertx, client.adapt(OpenShiftClient.class));
+            deploymentConfigOperations = new DeploymentConfigOperations(vertx, client.adapt(OpenShiftClient.class));
         } else {
-            imagesStreamResources = null;
+            imagesStreamOperations = null;
             buildConfigOperations = null;
+            deploymentConfigOperations = null;
         }
         this.zookeeperClusterOperations = new ZookeeperClusterOperations(vertx, isOpenShift, configMapOperations, serviceOperations, statefulSetOperations, pvcOperations);
         this.kafkaClusterOperations = new KafkaClusterOperations(vertx, isOpenShift, configMapOperations, serviceOperations, statefulSetOperations, pvcOperations);
-        this.kafkaConnectClusterOperations = new KafkaConnectClusterOperations(vertx, isOpenShift, configMapOperations, deploymentOperations, serviceOperations, imagesStreamResources, buildConfigOperations);
+        this.kafkaConnectClusterOperations = new KafkaConnectClusterOperations(vertx, isOpenShift, configMapOperations, deploymentOperations, serviceOperations);
+
+        if (isOpenShift) {
+            this.kafkaConnectS2IClusterOperations = new KafkaConnectS2IClusterOperations(vertx, isOpenShift, configMapOperations, deploymentConfigOperations, serviceOperations, imagesStreamOperations, buildConfigOperations);
+        } else {
```

Comment:
Do we need the else ? The `kafkaConnectS2IClusterOperations` field should be `null` by default and the it should be set only if `isOpenShift`.

Reply:
Yeah, that sounds reasonable. Fixed.